hawkbit-dmf-amqp/src/test/java/org/eclipse/hawkbit/RabbitMqSetupService.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit;
+
+import java.net.MalformedURLException;
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+
+import org.eclipse.hawkbit.amqp.AmqpAuthenticationMessageHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.amqp.RabbitProperties;
+import org.springframework.util.ReflectionUtils;
+import org.springframework.util.StringUtils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.rabbitmq.http.client.Client;
+import com.rabbitmq.http.client.domain.UserPermissions;
+
+/**
+ * Creates and deletes a new virtual host if the rabbit mq management api is
+ * available.
+ * 
+ *
+ */
+public class RabbitMqSetupService {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AmqpAuthenticationMessageHandler.class);
+
+    private static Client rabbitmqHttpClient;
+    private String virtualHost;
+
+    private final String hostname;
+
+    private String username;
+
+    private String password;
+
+    public RabbitMqSetupService(RabbitProperties properties) {
+        hostname = properties.getHost();
+        username = properties.getUsername();
+        password = properties.getPassword();
+    }
+
+    protected Client getRabbitmqHttpClient() {
+        if (rabbitmqHttpClient == null) {
+            try {
+                rabbitmqHttpClient = new Client("http://" + getHostname() + ":15672/api/", getUsername(),
+                        getPassword());
+            } catch (MalformedURLException | URISyntaxException e) {
+                ReflectionUtils.rethrowRuntimeException(e);
+            }
+        }
+        return rabbitmqHttpClient;
+    }
+
+    @PostConstruct
+    void createVirtualHost() {
+        try {
+            if (!getRabbitmqHttpClient().alivenessTest("/")) {
+                throw new AlivenessException(getHostname());
+
+            }
+        } catch (final AlivenessException e) {
+            ReflectionUtils.rethrowRuntimeException(e);
+        } catch (final Exception e) {
```

Comment:
How do you now that the cause for the exception is a connection issue? Better log the actual reason